### PR TITLE
DSND-2138: Finish implementation of NEWINC/SH01 child

### PR DESCRIPTION
**Pull request by sthompsonCH**
## Describe the changes
- Transform NEWINC delta with a child SH01 inside the child array into a valid PUT request object to be sent to the API.

### Related Jira tickets
[DSND-2138](https://companieshouse.atlassian.net/browse/DSND-2138)

### General
- [ ] Is the PR as small as it can be?
- [ ] Has the Jira ticket been updated with any relevant comments?
- [ ] Is the `README` up to date?
- [ ] Has the `POM` been updated for the latest versions of dependencies?
- [ ] Has the code been double-checked against `main`?
- [ ] Does the code adhere standards in this repository, including SonarQube checks?

### Testing
- [ ] Do the code changes have unit and integration tests with code coverage > 80%?
- [ ] Has the code been tested locally and/or passed the API Karate tests on Tilt?
- [ ] Have mandatory manual test cases been run?
- [ ] Are extra Karate tests required?
- [ ] Are all the test data resources up to date?

### Release preparation
_Where possible, add links to the respective Jira tickets when an item is checked_
- [ ] Have these changes been included in a release ticket?
- [ ] Are changes required to the `docker-chs-development` deployment or test data?
- [ ] Are any changes required to the environment added to `chs-configs`?

## Notes to the tester
_Add any testing hints or instructions here._

[DSND-2138]: https://companieshouse.atlassian.net/browse/DSND-2138?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/filinghistory/consumer/mapper/ChildNodeMapperFactory.java [hunk 1]
```diff
@@ -8,17 +8,20 @@ public class ChildNodeMapperFactory {
...
         this.annotationNodeMapper = annotationNodeMapper;
         this.associatedFilingNodeMapper = associatedFilingNodeMapper;
+        this.embeddedChildNodeMapper = embeddedChildNodeMapper;
     }
 
     public ChildNodeMapper getChildMapper(TransactionKindEnum kind) {
         return switch (kind) {
             case ANNOTATION -> annotationNodeMapper;
             case ASSOCIATED_FILING -> associatedFilingNodeMapper;
+            case TOP_LEVEL -> embeddedChildNodeMapper; // TODO: Check if TOP_LEVEL makes sense here?
```
Comment (cgregoryCH) [inline src/main/java/uk/gov/companieshouse/filinghistory/consumer/mapper/ChildNodeMapperFactory.java:24]: Makes sense to me, we only get here if its a parent (top level transaction) with a child embedded in it. If other edge cases arise in the future we can change it.

File: src/main/java/uk/gov/companieshouse/filinghistory/consumer/mapper/ChildNodeMapperFactory.java [hunk 2]
```diff
@@ -8,17 +8,20 @@ public class ChildNodeMapperFactory {
...
             default -> throw new IllegalStateException("Unexpected value: " + kind.getValue());
```
Comment (cgregoryCH) [inline src/main/java/uk/gov/companieshouse/filinghistory/consumer/mapper/ChildNodeMapperFactory.java:25]: This message could be more specific: "Unexpected transaction kind: ..."

File: src/main/java/uk/gov/companieshouse/filinghistory/consumer/mapper/EmbeddedChildNodeMapper.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+        ChildProperties embeddedChild = delta.getChild().getFirst();
+
+        ObjectNode objectNode = objectMapper.createObjectNode()
+                .put("type", embeddedChild.getFormType())
+                .put("date", formatDate.format(embeddedChild.getReceiveDate()))
+                .put("description",
+                        StringUtils.isNotBlank(embeddedChild.getDescription()) ? embeddedChild.getDescription() : "");
+
+        return new ChildPair("associated_filings", objectNode);
```
Comment (cgregoryCH) [inline src/main/java/uk/gov/companieshouse/filinghistory/consumer/mapper/EmbeddedChildNodeMapper.java:33]: All nodeMappers could have this field name be made a constant. Also, the ChildPair field name currently "type" could be more specific, something like childArrayKey? just so its obvious when its used the processor.

File: src/test/java/uk/gov/companieshouse/filinghistory/consumer/kafka/ConsumerPositiveComprehensiveIT.java [hunk 4]
```diff
@@ -67,6 +67,7 @@ public void setup() {
 
     @ParameterizedTest
     @CsvSource({
+            "363s/363s",
```
Comment (cgregoryCH) [inline src/test/java/uk/gov/companieshouse/filinghistory/consumer/kafka/ConsumerPositiveComprehensiveIT.java:70]: should be in the annual_return directory, using the mongo data.category field as the naming convention

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

File: src/test/java/uk/gov/companieshouse/filinghistory/consumer/kafka/ConsumerPositiveComprehensiveIT.java [hunk 5]
```diff
@@ -68,7 +68,11 @@ public void setup() {
     @ParameterizedTest
     @CsvSource({
             "annotations/annotation",
-            "associated_filing/associated_filing",
+
+            "annual_return/annual_return",
```
Comment (cgregoryCH) [inline src/test/java/uk/gov/companieshouse/filinghistory/consumer/kafka/ConsumerPositiveComprehensiveIT.java:72]: The naming convention for the other top level transactions are category/form_type. Could this be changed please? I think having child transaction ones in specific folders is fine though.

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Review by crowlandsCH — APPROVED** [review]

**Pull request merged**